cmd/root.go

```diff
@@ -165,29 +205,34 @@ func Execute(analyticsTracker analytics.Tracker, version string) {
 	}
 
 	// change the completion command help
-	rootCmd.InitDefaultCompletionCmd()
-	completionCmd, _, err := rootCmd.Find([]string{"completion"})
+	rootCmd.Cmd().InitDefaultCompletionCmd()
+	completionCmd, _, err := rootCmd.Cmd().Find([]string{"completion"})
 	if err == nil {
 		completionCmd.Long = fmt.Sprintf(`Generate the autocompletion script for %[1]s for the specified shell.
-See each command's help for details on how to use the generated script.`, rootCmd.Name())
-		rootCmd.AddCommand(completionCmd)
+See each command's help for details on how to use the generated script.`, rootCmd.Cmd().Name())
+		rootCmd.Cmd().AddCommand(completionCmd)
 	}
 
 	err = rootCmd.Execute()
-	outcome := analytics.SUCCESS
-	if err != nil {
+
+	var outcome string
+	switch {
+	case rootCmd.HelpCalled():
+		outcome = analytics.HELP
+	case err != nil:
 		outcome = analytics.ERROR
 		fmt.Fprintln(os.Stderr, err.Error())
+	default:
+		outcome = analytics.SUCCESS
 	}
 
-	optOutStr := rootCmd.PersistentFlags().Lookup(cliflags.AnalyticsOptOut).Value.String()
-	optOut, _ := strconv.ParseBool(optOutStr)
-	analyticsTracker.SendCommandCompletedEvent(
-		rootCmd.PersistentFlags().Lookup(cliflags.AccessTokenFlag).Value.String(),
-		rootCmd.PersistentFlags().Lookup(cliflags.BaseURIFlag).Value.String(),
-		optOut,
-		outcome,
+	analyticsClient := trackerFn.Tracker(version)(
+		rootCmd.cmd.PersistentFlags().Lookup(cliflags.AccessTokenFlag).Value.String(),
+		rootCmd.cmd.PersistentFlags().Lookup(cliflags.BaseURIFlag).Value.String(),
+		viper.GetBool(cliflags.AnalyticsOptOut),
```

Comment:
I double-checked that this was already set, and it is, so we don't need to get it the command's flags.